### ISSUE TITLE
False negative on `UnnecessaryLet` in nested nullable property

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6373]
## Expected Behavior
Following TC should pass

```kotlin
@Test
fun `reports nested nullable property`() {
    val findings = subject.compileAndLintWithContext(
        env,
        """
            class Dialog(val window: Int?)
            val dialog: Dialog? = null
            fun test() {
                dialog?.let {
                    it.window?.let {
                        it++
                        println(it)
                    }
                }
            }
    """.trimIndent()
    )
    assertThat(findings).hasSize(1)
}
```

Compliant code should be as below

```kotlin
class Dialog(val window: Int?)
val dialog: Dialog? = null
fun test() {
    dialog?.window?.let {
        it++
        println(it)
    }
}
```

## Observed Behavior
Above TC is failing

## Context
There is existing [issue](https://github.com/detekt/detekt/issues/6222) to create a new rule to detect this but I think this should be handled by the `UnnecessaryLet`

**Comment by BraisGabin** [comment]
Agree

**Comment by mjovanc** [comment]
I could try looking into this if someone hasn't already started.

**Comment by atulgpt** [comment]
Hi @mjovanc you can go ahead. I am not looking into this

**Comment by cortinico** [comment]
> I could try looking into this if someone hasn't already started.

That's assigned to you to work on it if you wish @mjovanc 👍

**Comment by mjovanc** [comment]
@cortinico Will do! Thanks!

**Comment by atulgpt** [comment]
@BraisGabin @cortinico I have raised the PR for fixing the issue

**Comment by atulgpt** [comment]
@cortinico can you take another look. thanks
